features/frameworks/mbed-coap/source/sn_coap_protocol.c

```diff
@@ -363,6 +363,36 @@ int8_t sn_coap_protocol_delete_retransmission(struct coap_s *handle, uint16_t ms
     return -2;
 }
 
+int8_t sn_coap_protocol_delete_retransmission_by_token(struct coap_s *handle, uint8_t *token, uint8_t token_len)
+{
+#if ENABLE_RESENDINGS /* If Message resending is not used at all, this part of code will not be compiled */
+    if (handle == NULL || token == NULL || token_len == 0) {
+        tr_error("sn_coap_protocol_delete_retransmission_by_token NULL");
+        return -1;
+    }
+
+    ns_list_foreach(coap_send_msg_s, stored_msg, &handle->linked_list_resent_msgs) {
+        uint8_t stored_token_len =  (stored_msg->send_msg_ptr->packet_ptr[0] & 0x0F);
+        if (stored_token_len == token_len) {
+            uint8_t stored_token[8];
+            memcpy(stored_token, &stored_msg->send_msg_ptr->packet_ptr[4], stored_token_len);
+            if (memcmp(stored_token, token, stored_token_len) == 0) {
+                uint16_t temp_msg_id = (stored_msg->send_msg_ptr->packet_ptr[2] << 8);
+                temp_msg_id += (uint16_t)stored_msg->send_msg_ptr->packet_ptr[3];
+                tr_debug("sn_coap_protocol_delete_retransmission_by_token - removed msg_id: %d", temp_msg_id);
+                ns_list_remove(&handle->linked_list_resent_msgs, stored_msg);
+                --handle->count_resent_msgs;
+
+                /* Free memory of stored message */
+                sn_coap_protocol_release_allocated_send_msg_mem(handle, stored_msg);
+                return 0;
+            }
+        }
+    }
+#endif
+    return -2;
```

Comment:
Is it correct that if if the #define is not enabled, this function will always return an error?

Reply:
Yes, because this function handles cases where re-sending of messages is causing internal queue to be full hence there is a methodology to remove those messages. However, if the resending is not enabled then this function will always return error.

Reply:
Just wanted to be sure, since the pattern looked strange.